plugins/metal/plugin.go

```diff
@@ -0,0 +1,218 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: MIT
+
+package metal
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/netip"
+	"os"
+	"strings"
+
+	"github.com/coredhcp/coredhcp/handler"
+	"github.com/coredhcp/coredhcp/logger"
+	"github.com/coredhcp/coredhcp/plugins"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv6"
+	"github.com/ironcore-dev/fedhcp/internal/api"
+	"github.com/ironcore-dev/fedhcp/internal/kubernetes"
+	ipamv1alpha1 "github.com/ironcore-dev/ipam/api/ipam/v1alpha1"
+	metalv1alpha1 "github.com/ironcore-dev/metal-operator/api/v1alpha1"
+	"github.com/mdlayher/netx/eui64"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+var log = logger.GetLogger("plugins/metal")
+
+var Plugin = plugins.Plugin{
+	Name:   "metal",
+	Setup6: setup6,
+	Setup4: setup4,
+}
+
+// map MAC address to machine name
+var machineMap map[string]string
+
+// args[0] = path to configuration file
+func parseArgs(args ...string) (string, error) {
+	if len(args) != 1 {
+		return "", fmt.Errorf("exactly one argument must be passed to the metal plugin, got %d", len(args))
+	}
+	return args[0], nil
+}
+
+func setup6(args ...string) (handler.Handler6, error) {
+	var err error
+	machineMap, err = loadConfig(args...)
+	if err != nil {
+		return nil, err
+	}
+
+	return handler6, nil
+}
+
+func loadConfig(args ...string) (map[string]string, error) {
+	log.Info("Loading metal config")
+	path, err := parseArgs(args...)
+	if err != nil {
+		return nil, fmt.Errorf("invalid configuration: %v", err)
+	}
+
+	log.Info("Reading metal config file", "ConfigFile", path)
+	configData, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config file: %v", err)
+	}
+
+	var config []api.Machine
+	if err = json.Unmarshal(configData, &config); err != nil {
```

Comment:
Can we use `yaml.Marshal`-ing here instead of a JSON based inventory format.

Reply:
Sure, let's go for YAML, it's much more readable